integration/bootstrap_before_buffer_rotation_no_tick_regression_test.go

```diff
@@ -0,0 +1,207 @@
+	// test bootstrapper completes, the commitlog bootstrapper will run.
+	bootstrapOpts := newDefaulTestResultOptions(setup.storageOpts)
+	bootstrapCommitlogOpts := bcl.NewOptions().
+		SetResultOptions(bootstrapOpts).
+		SetCommitLogOptions(commitLogOpts)
+	commitlogBootstrapper, err := bcl.NewCommitLogBootstrapper(bootstrapCommitlogOpts, nil)
+	require.NoError(t, err)
+
+	// Setup the test bootstrapper to only return success when a signal is sent.
+	signalCh := make(chan struct{})
+	test := newTestBootstrapperSource(testBootstrapperSourceOptions{
+		read: func(_ namespace.Metadata, shardTimeRanges result.ShardTimeRanges, _ bootstrap.RunOptions) (result.BootstrapResult, error) {
+			<-signalCh
+			result := result.NewBootstrapResult()
+			// Mark all as unfulfilled so the commitlog bootstrapper will be called after
+			result.SetUnfulfilled(shardTimeRanges)
+			return result, nil
+		},
+	}, bootstrapOpts, commitlogBootstrapper)
+
+	process := bootstrap.NewProcess(test, bootstrapOpts)
+	setup.storageOpts = setup.storageOpts.SetBootstrapProcess(process)
+
+	// Start a background goroutine which will wait until the server is started,
+	// issue a single write into the active block, change the time to be far enough
+	// into the *next* block that the series buffer for the previously active block
+	// can be rotated, and then signals the test bootstrapper that it can proceed.
+	go func() {
+		// Wait for server to start
+		setup.waitUntilServerIsUp()
+
+		// Set the time such that the (previously) active block is ready to be flushed.
+		now = now.Add(blockSize).Add(ropts.BufferPast()).Add(time.Second)
+		setup.setNowFn(now)
+
+		// Set the tick interval to be so large we can "hang" a tick at the end, preventing
+		// it from completing until we're ready to "resume" it later.
+		runtimeMgr := setup.storageOpts.RuntimeOptionsManager()
+		existingOptions := runtimeMgr.Get()
+		newOptions := existingOptions.SetTickMinimumInterval(2 * time.Hour)
+
+		// Sleep for a multiple of the old tick interval to guarantee that the "hung" tick
+		// that we will trigger after has a "tick start" time equal to the value of "now"
+		// above.
+		time.Sleep(existingOptions.TickMinimumInterval() * 10)
+
+		// Update the runtime minimumTickInterval value
+		err := runtimeMgr.Update(newOptions)
+		if err != nil {
+			panic(err)
+		}
+
+		// Sleep for a multiple of the old tick interval to guarantee that we enter a "hung"
+		// tick.
+		time.Sleep(existingOptions.TickMinimumInterval() * 10)
+
+		// Twice because the test bootstrapper will need to run two times, once to fulfill
+		// all historical blocks and once to fulfill the active block.
+		signalCh <- struct{}{}
+		signalCh <- struct{}{}
+	}()
+	require.NoError(t, setup.startServer())
+
+	// Now that bootstrapping has completed, re-enable ticking so that flushing can take place
+	setup.mustSetTickMinimumInterval(100 * time.Millisecond)
+
+	// Wait for a flush to complete
+	setup.sleepFor10xTickMinimumInterval()
```

Comment:
do you want to add a waitgroup/channel to ensure the goroutine above has finished before doing the assertions below?

Reply:
dont need one. startServer wont return until bootstrap completes, and bootstrap cant complete until that background routine completes

Reply:
cool, mind putting a note down by the startServer call to capture that

Reply:
done